File: services/clean-code/internal/ast/parser/testdata/java/Sample.java
// Fixture module for the clean-code v1 Java parser adapter.
// Stage 2.1 tests assert the parser extracts the canonical AST
// scopes from this file (file, package, one interface, one
// class with two methods).

package com.example.sample;

import java.util.ArrayList;
import java.util.List;

public interface Sample {
    String sample(int seed);

    void close();
}

class MemorySample implements Sample {
    private final List<String> values;
    private boolean closed;

    public MemorySample(List<String> initial) {
        this.values = new ArrayList<>(initial);
        this.closed = false;
    }

    public String sample(int seed) {
        if (closed) {
            throw new IllegalStateException("sample: already closed");
        }
        if (values.isEmpty()) {
            throw new IllegalStateException("sample: empty buffer");
        }
        return values.get(seed % values.size());
    }

    public void close() {
        closed = true;
    }
}

File: services/clean-code/internal/ast/parser/testdata/python/sample.py
"""Fixture module for the clean-code v1 Python parser adapter.

Stage 2.1 tests assert the parser extracts the canonical AST
scopes from this file (file, two functions, one class with two
methods, one ABC interface).
"""

from abc import ABC, abstractmethod
from typing import List, Optional


class Sampler(ABC):
    """ABC base every concrete sampler implements."""

    @abstractmethod
    def sample(self, seed: int) -> str:
        ...

    @abstractmethod
    def close(self) -> None:
        ...


class MemorySampler:
    """Concrete sampler backed by an in-memory list."""

    def __init__(self, initial: Optional[List[str]] = None) -> None:
        self._values: List[str] = list(initial or [])
        self._closed: bool = False

    def sample(self, seed: int) -> str:
        if self._closed:
            raise RuntimeError("sample: already closed")
        if not self._values:
            raise RuntimeError("sample: empty buffer")
        return self._values[seed % len(self._values)]

    def close(self) -> None:
        self._closed = True


def make_sampler(initial: Optional[List[str]] = None) -> MemorySampler:
    return MemorySampler(initial)


async def async_make_sampler(initial: Optional[List[str]] = None) -> MemorySampler:
    return MemorySampler(initial)
